set-web/src/main/java/eu/europeana/set/web/service/impl/UserSetServiceImpl.java

```diff
@@ -365,6 +371,59 @@ private void validateControlledValues(UserSet webUserSet) throws RequestBodyVali
 	}
     }
     
+    /**
+     * The value of isDefinedBy is validated (e.g. search URL https://api.europeana.eu/record/search.json?) 
+     * to point to the Search API.
+     * We make a GET request upon creation to see if the request returns a 200 (meaning is valid).
+     * @param webUserSet
+     * @throws ParamValidationException 
+     * @throws RequestBodyValidationException
+     */
+    private void validateSearchUri(UserSet webUserSet) 
+	    throws ParamValidationException, RequestBodyValidationException {
+	
+	int STATUS_OK = 200;
+
+	if (webUserSet.getIsDefinedBy() != null) {
+            String searchUrl = getBaseSearchUrl(getConfiguration().getSearchApiUrl());
+            String queryUrl = getBaseSearchUrl(webUserSet.getIsDefinedBy());
+            if (!queryUrl.equals(searchUrl)) {	    
```

Comment:
please change the order of variables to avoid NPE

Reply:
order changed